analyzer/src/main/kotlin/curation/Sw360PackageCurationProvider.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.curation
+
+import com.fasterxml.jackson.databind.DeserializationFeature
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.eclipse.sw360.antenna.http.HttpClientFactoryImpl
+import org.eclipse.sw360.antenna.http.config.HttpClientConfig
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360Connection
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ConnectionFactory
+import org.eclipse.sw360.antenna.sw360.client.config.SW360ClientConfig
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.attachments.SW360AttachmentType
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360ClearingState
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360Release
+
+import org.ossreviewtoolkit.analyzer.PackageCurationProvider
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.HashAlgorithm
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.PackageCuration
+import org.ossreviewtoolkit.model.PackageCurationData
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.config.Sw360StorageConfiguration
+import org.ossreviewtoolkit.model.jsonMapper
+
+/**
+ * A [PackageCurationProvider] for curated package meta-data from the configured SW360 instance using the REST API.
+ */
+class Sw360PackageCurationProvider(sw360Configuration: Sw360StorageConfiguration?) : PackageCurationProvider {
+
+    init {
+        requireNotNull(sw360Configuration) {
+            "SW360 is not configured. To use curations from SW360, configure it in the ORT configuration file."
+        }
+    }
+
+    private val sw360JsonMapper = jsonMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+    private val sw360ConnectionFactory = createSw360Connection(sw360Configuration!!, sw360JsonMapper)
+    private val sw360ComponentClient = sw360ConnectionFactory.componentAdapter
+    private val sw360ReleaseClient = sw360ConnectionFactory.releaseAdapter
```

Comment:
It is probably a matter of taste, but I prefer not having too many member fields. The sw360JsonMapper is certainly unnecessary. With regards to the clients, you can consider storing only the connection and access them from there when needed.